samples/data-sealing/Makefile

```diff
@@ -1,24 +1,91 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
+include ../config.mk
+
 .PHONY: all build clean run
 
-OE_CRYPTO_LIB := mbedtls
-export OE_CRYPTO_LIB
+OE_CRYPTO_LIB?= mbedtls
+OE_SDK_BUILD_TYPE?= Release RelWithDebInfo MinSizeRel Debug
+
+EDL:= datasealing
+
+define oe-config
+$(or $(shell pkg-config oe$(strip $1)),	\
+	$(error Have you sourced 'openenclaverc'?))
+endef
+
+define find-seal-plugins
+$(wildcard $(LIBDIR)/enclave/objects-$1/oeseal_*/*.o)
```

Comment:
Currently the PR installs .o files into the installed path `l/opt/openenclave/lib/openenclave/enclave/objects-Debug`

When the OE is built with LVI mitigation, the objects from both following paths will be pulled with make, which causes failures
```
/opt/openenclave/lib/openenclave/enclave/objects-Debug/oeseal_gcmaes/seal_gcmaes.c.o
opt/openenclave/lib/openenclave/enclave/objects-Debug/oeseal_gcmaes-lvi-cfg/seal_gcmaes.c.o
```

Reply:
@mingweishih, the reason to install a .o instead of .a file is to keep the constructor (https://github.com/binxing/openenclave/blob/pr/sealing/enclave/sealing/sgx/seal_gcmaes.c#L267) so that plug-ins can be auto-registered. We discussed this in SIG-Arch and @CodeMonkeyLeet was ok with it if I wasn't mistaken.

Reply:
Sorry that I didn't follow the discussion. If there is already an agreement of using objects, then please keep it. However, I would still suggest
- Install the object to the path: `/opt/openenclave/lib/openenclave/enclave/objects/` regardless it's debug or release build
- Use the naming: `seal_gcmaes.c.o` and `seal_gcmaes-lvi-cfg.c.o` to differentiate the normal and LVI-mitigated versions.
  Ideally, I'd like to have just `seal_gcmaes.o` and `seal_gcmaes-lvi-cfg.o`, which looks cleaner.

Reply:
One option to consider for **future** would be whether we can use `--whole-archive` and `--no-whole-archive` pair to ensure that all the objects in a plugin are included in an enclave.

I'm not asking that we change the agreement on using object file; just something to consider for future.